msal/token_cache.py

```diff
@@ -150,8 +151,9 @@ def _find(self, credential_type, target=None, query=None):  # O(n) generator
                     if entry != preferred_result:  # Avoid yielding the same entry twice
                         yield entry
 
-    def find(self, credential_type, target=None, query=None):  # Obsolete. Use _find() instead.
-        return list(self._find(credential_type, target=target, query=query))
+    def find(self, credential_type, target=None, query=None):
+        warnings.warn("Use list(search()) instead of find().", DeprecationWarning)
```

Comment:
If `msal.token_cache.TokenCache.find` is deprecated, MSAL itself should not use it anymore:

https://github.com/AzureAD/microsoft-authentication-library-for-python/blob/48a7bd3c8e7040c02165e4e12f3a8004f18391a3/msal/application.py#L1111-L1113

Reply:
Good catch! Github's reference feature somehow did not show me those old `find()` references.

In the new commit, most of the `find()` have been replaced with `search()`, but I also remove the deprecation warning. Because in some circumstances, the caller would actually want a list, not a generator.

Reply:
Good!